src/test/ui/lto/auxiliary/thinlto-dylib.rs

```diff
@@ -0,0 +1,23 @@
+// Auxiliary crate for test issue-105637: the LTOed dylib which had duplicate symbols from libstd,
+// breaking the panic hook feature.
+//
+// This simulates the `rustc_driver` crate, and the main crate simulates rustc's main binary hooking
+// into this driver.
+
+// compile-flags: -Zdylib-lto -C lto=thin
```

Comment:
Doesn't this need `--crate-type dylib` or does compiletest already pass it?

Reply:
compiletest passes it yes:

`"./build/x86_64-unknown-linux-gnu/stage1/bin/rustc" "./src/test/ui/lto/auxiliary/thinlto-dylib.rs" "-Zthreads=1" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Zdeduplicate-diagnostics=no" "-Cstrip=debuginfo" "-C" "prefer-dynamic" "--out-dir" "./build/x86_64-unknown-linux-gnu/test/ui/lto/issue-105637/auxiliary" "-Crpath" "-Cdebuginfo=0" "-Lnative=./build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-Zdylib-lto" "-C" "lto=thin" "--crate-type" "dylib" "-L" "./build/x86_64-unknown-linux-gnu/test/ui/lto/issue-105637/auxiliary"`

Reply:
Looks like there are some targets where it doesn't build as dylib: https://github.com/rust-lang/rust/blob/49c2279ef658d8732597c4da93897d84838f3df5/src/tools/compiletest/src/runtest.rs#L1744 This seems to be mostly targets where dylibs aren't supported anyway. I think this is fine.

Reply:
We may need to protect the test with `// no-prefer-dynamic` then. Since CI is already running, I'll do that if it causes a failure.